gotradepal/backend/app/main.py

```
@@ -0,0 +1,73 @@
+    return {"status": "ok", "service": "gotradepal"}
+
+@app.post("/auth/signup", response_model=Token)
+def signup(payload: UserCreate, db: Session = Depends(get_db)):
+    if db.scalar(select(User).where(User.email == payload.email)):
+        raise HTTPException(409, "Email already registered")
+    user = User(name=payload.name, email=payload.email, phone=payload.phone, country=payload.country, hashed_password=hash_password(payload.password))
+    db.add(user); db.commit(); db.refresh(user)
+    return Token(access_token=create_access_token(str(user.id)))
+
+@app.post("/auth/login", response_model=Token)
+def login(email: str, password: str, db: Session = Depends(get_db)):
```

Ruff (0.15.21)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)